File: app/src/main/java/com/example/test/mvvmsampleapp/service/repository/GitHubService.java
package com.example.test.mvvmsampleapp.service.repository;

import com.example.test.mvvmsampleapp.service.model.Project;

import java.util.List;

import io.reactivex.Single;
import retrofit2.http.GET;
import retrofit2.http.Path;

interface GitHubService {
    String HTTPS_API_GITHUB_URL = "https://api.github.com/";

    @GET("users/{user}/repos")
    Single<List<Project>> getProjectList(@Path("user") String user);

    @GET("/repos/{user}/{reponame}")
    Single<Project> getProjectDetails(@Path("user") String user, @Path("reponame") String projectName);
}

File: app/src/main/java/com/example/test/mvvmsampleapp/service/repository/GitHubService.kt
package com.example.test.mvvmsampleapp.service.repository

import com.example.test.mvvmsampleapp.service.model.Project

import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Path

/**
 * Network service with a list of all network calls in the app via retrofit.
 */
interface GitHubService {

    @GET("users/{user}/repos")
    fun getProjectList(@Path("user") user: String): Call<List<Project>>

    @GET("repos/{user}/{reponame}")
    fun getProjectDetails(@Path("user") user: String, @Path("reponame") projectName: String): Call<Project>

    companion object {
        val HTTPS_API_GITHUB_URL = "https://api.github.com/"
    }

}
